### DISCUSSION TITLE
support wlroots and support swaywm

### DISCUSSION THREAD

**Discussion by xzl01** [linuxdeepin/developer-center#3876] [Features Request & Ideas | 特性请求 & 头脑风暴]
swaywm 是一个wayland平铺窗管，wlroot是sway团队开发的一个用于Wayland开发的模块，我期望能在v23中支持swaywm和wlroots

**Comment by xzl01** [comment]
# wlroots

- [x] libavformat-dev
- [x] libavcodec-dev
- [x] libcap-dev
- [x] libdrm-dev 版本过低需要升级
- [x] libgdm-dev
- [x] libxcb-composite0-dev
- [x] libxcb-dri3-dev
- [x] libxcb-icccm4-dev
- [x] libxcb-image0-dev
- [x] libxcb-present-dev
- [x] libxcb-render0-dev
- [x] libxcb-render-util0-dev
- [x] libxcb-res0-dev
- [x] libxcb-xfixes0-dev
- [x] libgbm-dev
- [x] libgles2-mesa-dev
- [x] libpixman-1-dev
- [x] libseat-dev
- [x] libsystemd-dev
- [x] libwayland-dev
- [x] libxcb-xinput-dev
- [x] libx11-xcb-dev meson
## 额外的支持

- [x] glslang-tools

# libdrm-dev

- [x] quilt
- [x] xsltproc
- [x] python3-setuptools
- [x] xutils-dev
- [x] python3-docutils
  
# libseat-dev

- [x] scdoc

# sway

- [x] libcairo2-dev
- [x] libdbus-1-dev
- [x] libgdk-pixbuf2.0-dev
- [x] libjson-c-dev
- [x] libpam0g-dev
- [x] libpango1.0-dev
- [x] pkgconf
- [x] tree
- [x] libwlroots-dev
- [x] swaybg
  

# swaybg

- [x] wayland-protocols > 1.26 (需要升级)

**Comment by xzl01** [comment]
[wlroots](https://www.aliyundrive.com/s/gKYLJJmLKYQ) 链接

**Comment by mchnjt** [comment]
swaywm 不是窗口管理器吗？ deepin已经有一个基于kwin的

**Reply by BLumia** [reply]
是，这种需求上不冲突。

**Comment by wineee** [comment]
进展如何？

**Reply by xzl01** [reply]
没有给出明确决议 我在自己电脑上已经可以正常运行 待有决议方案后继续打包

**Reply by xzl01** [reply]
0.15版本已经进入ci仓库 现打包0.16版本集成

**Comment by Zeno-sole** [comment]
@xzl01  我这边的意见是更新 gdm mesa xcb drm等相关组件要了解影响范围，最好自行验证一下，确保不出现严重问题即可。

**Comment by BLumia** [comment]
我这边的意见与 zeno 的一样，更新 xcb 之类包的时候确保没问题就可以。

鉴于计划内的新 DDE Wayland 用窗管基于 wlroots，所以 wlroots 是肯定要支持的。

**Comment by xzl01** [comment]
计划更新wlroots至0.16版本
